pkg/providers/instance/instance.go

```diff
@@ -221,9 +218,10 @@ func (p *Provider) newNetworkInterfaceForVM(vmName string, backendPools *loadbal
 					Properties: &armnetwork.InterfaceIPConfigurationPropertiesFormat{
 						Primary:                   to.Ptr(true),
 						PrivateIPAllocationMethod: to.Ptr(armnetwork.IPAllocationMethodDynamic),
-						Subnet: &armnetwork.Subnet{
-							ID: &p.subnetID,
-						},
+						// Subnet is set in p.applyTemplateToNic
```

Comment:
Really its just because this is the only bit of networking configuration defined elsewhere, I figured an explicit callout would be good. 

We can remove if you think where subnet is set is discoverable enough. This comment isn't great anyway because it doesn't communicate the logic for how its set and exactly where in the launch template provider its set maybe we can remove. 


By typing that all I convinced myself to remove it.